shared/src/main/scala/ListHelpers.scala

```diff
@@ -5,6 +5,28 @@ import VNum.given
 
 object ListHelpers:
 
+  def filter(iterable: VList, predicate: VFun)(using ctx: Context): VList =
+    val branches = predicate.originalAST match
+      case Some(lam) => lam.body
+      case None =>
+        return VList(iterable.zipWithIndex.map { (item, index) =>
+          predicate.execute(item, index, List(item))
+        }*)
+
+    val filtered = iterable.zipWithIndex.filter((item, index) =>
+      var temp = true
+      for branch <- branches do
+        temp = temp && MiscHelpers.boolify(
+          VFun
+            .fromLambda(AST.Lambda(1, List.empty, List(branch)))
+            .execute(item, index, List(item))
+        )
+      temp
+    )
```

Comment:
```suggestion
    val filtered = iterable.zipWithIndex.filter { (item, index) =>
      branches.forall { branch =>
        MiscHelpers.boolify(
          VFun
            .fromLambda(AST.Lambda(1, List.empty, List(branch)))
            .execute(item, index, List(item))
        )
      }
    }
```
The braces are just a personal style thing, we can keep the parens if you want.